apps/accounts/views.py

```diff
@@ -132,3 +132,16 @@ class ProfileView(JWTProtectedView, generics.RetrieveAPIView):
 
     def retrieve(self, request, *args, **kw):
         return Response(self.get_serializer(request.user).data)
+
+
+class AccountSourceView(generics.RetrieveAPIView):
+    serializer_class = AccountSourceSerializer
+
+    @waffle_switch('fxa-auth')
+    def retrieve(self, request, *args, **kwargs):
+        email = request.GET.get('email')
+        try:
+            user = UserProfile.objects.get(email=email)
+        except UserProfile.DoesNotExist:
+            user = UserProfile(fxa_id='sure')
```

Comment:
It sets `source: fxa` so that you can't tell if the account exists once the user has migrated. This way you can only discover if an account exists and hasn't migrated, not that the account exists.